openjdk/build.xml

```diff
@@ -24,42 +24,53 @@
 	<property name="DEST" value="${BUILD_ROOT}/openjdk" />
 	<property name="src" location="." />
 	<property environment="env" />
-	<property name="LIB" value="tohandler_simple"/>
+	
+	<if>
+		<or>
+		    	<matches pattern="^[89]{1}$" string="${JDK_VERSION}"/>
+			<matches pattern="^[1][01234]" string="${JDK_VERSION}"/>
+		</or>
+	 	<then>
+	    		<property name="jtregTar" value="jtreg_4_2_0_tip"/>
+		</then>
+		<else>
+	   		 <property name="jtregTar" value="jtreg_5_1_b01"/>
+		</else>
+	</if>
+	<if>
+	 	<or>
+	    		<equals arg1="${JDK_IMPL}" arg2="ibm"  />
+	    		<equals arg1="${JDK_IMPL}" arg2="openj9" />
+		</or>
+		<then>
+		    	<property name="openj9jtregtimeouthandler" value=",tohandler_simple"/>
+		</then>
+		<else>
+		 	<property name="openj9jtregtimeouthandler" value=""/>
+	       </else>
+	</if>
+	
+	<property name="LIB" value="${jtregTar}${openj9jtregtimeouthandler}"/>
+	
 	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml"/>
-
+	
 	<target name="getJtreg">
 		<mkdir dir="${DEST}"/>
-		<if>
-			<or>
-				<matches pattern="^[89]{1}$" string="${JDK_VERSION}"/>
-				<matches pattern="^[1][01234]" string="${JDK_VERSION}"/>
-			</or>
-			<then>
-				<property name="jtregTar" value="jtreg-4.2.0-tip.tar"/>
-			</then>
-			<else>
-				<property name="jtregTar" value="jtreg-5.1-b01.tar"/>
-			</else>
-		</if>
 		<if>
 			<not>
-				<available file="${jtregTar}" />
+				<available file="${LIB_DIR}/${jtregTar}.tar.gz" />
 			</not>
 			<then>
-				<if> <available file="custom_jtreg.tar.gz"/>
+				<if> 
+					<available file="custom_jtreg.tar.gz"/>
 					<then>
 						<echo message="Using custom_jtreg.tar.gz"/>
-							<copy file="custom_jtreg.tar.gz" tofile="${jtregTar}.gz"/>
+							<copy file="custom_jtreg.tar.gz" tofile="${jtregTar}.tar.gz"/>
 					</then>
-				<else>
-					<echo message="Getting Jtreg from https://ci.adoptopenjdk.net/job/dependency_pipeline/lastSuccessfulBuild/artifact/jtreg/${jtregTar}.gz"/>
-					<exec executable="curl" failonerror="true">
-						<arg line="--retry 5 -OLJSks https://ci.adoptopenjdk.net/job/dependency_pipeline/lastSuccessfulBuild/artifact/jtreg/${jtregTar}.gz" />
-					</exec>
-				</else>
 				</if>
```

Comment:
Copy `${LIB_DIR}/${jtregTar}.tar.gz` to current dir.

Reply:
So after the changes, line 68 should look like <copy file="${LIB_DIR}/${jtregTar}.tar.gz" tofile="${jtregTar}.tar.gz"/> if I'm not wrong?

Reply:
No, line 68 is correct. You need additional code to copy jtregTar to the current dir.

Reply:
I'm not able to understand what code should I add and exactly where

Reply:
If there is `${LIB_DIR}/${jtregTar}.tar.gz`, move it to current dir. That is, add else block in line 75:

```
<else>
   <copy file="${LIB_DIR}/${jtregTar}.tar.gz" tofile="${jtregTar}.tar.gz"/>
</else>
```

Reply:
Okay now I get it! I wish I could've figured this out before. Updated the code with the else block.

Reply:
Thanks for working on this. This PR is not an easy one :)

Reply:
Thank you so much for helping me out whenever I get stuck. I really enjoyed working on this PR!